[WIP] Surface the Model Names  on the Templates

maysunfaisal opened the pull request:
### What does this PR do?:
<!-- _Summarize the changes_ -->
- Surfaces the models being used in the template
- Provides a brief statement about the model being used

### Which issue(s) this PR fixes:
<!-- _Link to Github/JIRA issue(s)_ -->
https://issues.redhat.com/browse/DEVAI-56

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened and linked to this PR, if they are not in the PR scope due to various constraints.

- [x] Tested and Verified

  <!-- _I have verified that the changes were tested manually_ -->

- [ ] Documentation (READMEs, Product Docs, Blogs, Education Modules, etc.)

   <!-- _This includes READMEs, Product Docs, Blogs, Education Modules, etc._ -->


### How to test changes / Special notes to the reviewer:
Grab the template.yaml and import it to your RHDH to check out the Template wizard

Jdubrick reviewed (commented):
Changes look good to me. Does UX need to sign off on this?

![Screenshot 2024-09-30 at 12 36 25 PM](https://github.com/user-attachments/assets/da57ccb1-5def-4884-ae7a-6923562a5410)
![Screenshot 2024-09-30 at 12 36 33 PM](https://github.com/user-attachments/assets/6256a504-8ab8-4432-9302-e0318133c5b9)
![Screenshot 2024-09-30 at 12 36 44 PM](https://github.com/user-attachments/assets/02fb5a6b-3274-4257-8050-5c3caccf8cfe)
![Screenshot 2024-09-30 at 12 36 51 PM](https://github.com/user-attachments/assets/1be32aef-5ead-43cf-a3fc-d59aa61deb3b)

maysunfaisal commented:
@Jdubrick yes, I want to get a review from the UX, I will also add something more for the model servers.

thepetk reviewed (commented):
lgtm too :) 

I agree to what is mentioned above that the UX needs to approve those changes first :)

maysunfaisal commented:
@MariaLeonova could you please take a look?

MariaLeonova commented:
Thank you for tagging me! 

I have a question: where do the **model descriptions** come from? Are they being pulled from Hugging Face or is it something we can pre-determine and change? And if so, can links be put in descriptions?

Do users get a different list of models based on the model server they choose?

If possible, I'd suggest making the descriptions more concise and highlighting the intended use (i.e. Object Detection or Text Generation), so users are not overwhelmed with names and acronyms. Depending on the answer to my question, we can decide what info exactly to provide.

maysunfaisal commented:
@MariaLeonova 

> I have a question: where do the model descriptions come from? Are they being pulled from Hugging Face or is it something we can pre-determine and change? And if so, can links be put in descriptions?

The descriptions are from either Hugging Face or Github, or a combination of both. Do you mean to put the actual link on the UI description field? I currently have listed the source links in all the `scripts/envs/*` files just for our benefit

> Do users get a different list of models based on the model server they choose?

Currently all the model servers only have one model deployed with our Template but in the future yes, a server can have multiple models deployed and it would be nice to choose which model to use from the dropdown

> If possible, I'd suggest making the descriptions more concise and highlighting the intended use (i.e. Object Detection or Text Generation), so users are not overwhelmed with names and acronyms. Depending on the answer to my question, we can decide what info exactly to provide.

I am down to change the text description to be more helpful. Right now I did a direct copy-paste from the Hugging Face or GitHub pages. Unfortunately some of them have not so great introduction or documentation.

Jdubrick commented:
@maysunfaisal is this PR good on the UX side and ready for re-review? I can review this for sprint end if so

maysunfaisal commented:
@Jdubrick yes Jordan its good for a re-review

maysunfaisal commented:
@MariaLeonova could you please take a look at my recent commits? I have addressed the feedback:

<img width="1084" alt="Screenshot 2024-10-07 at 4 11 13 PM" src="https://github.com/user-attachments/assets/bd185b2e-31d3-4062-ba38-f44d5eebdae6">
<img width="1546" alt="Screenshot 2024-10-07 at 4 33 50 PM" src="https://github.com/user-attachments/assets/f4bcba40-5b97-4982-b711-56234cf0cc7b">
<img width="1075" alt="Screenshot 2024-10-07 at 4 47 18 PM" src="https://github.com/user-attachments/assets/c6868594-0ed3-4431-9b8e-1ceeeb11d9fc">

Jdubrick commented on scripts/envs/audio-to-text:
```diff
@@ -6,8 +6,21 @@ export APP_DISPLAY_NAME="Audio to Text Application"
+# https://github.com/containers/ai-lab-recipes/blob/main/model_servers/whispercpp/base/Containerfile
+# https://github.com/ggerganov/whisper.cpp/blob/v1.5.4/examples/server/README.md
 export MODEL_SERVICE_CONTAINER="quay.io/redhat-ai-dev/whispercpp:latest"
+export MODEL_SERVICE_DESC="Simple http server where WAV Files are passed to the inference model via http requests" 
```
Should we capitalize `HTTP` here?

maysunfaisal replied:
yes that makes sense

Jdubrick reviewed (approved):
looks good on my end

MariaLeonova commented:
@maysunfaisal This looks great, thank you very much for making the edits!
